MatrixGame/CMakeLists.txt

```diff
@@ -77,6 +70,7 @@ if(MATRIXGAME_BUILD_DLL)
     add_library(MatrixGame SHARED)
     # remove lib prefix from the DLL name (for GCC build)
     set_target_properties(MatrixGame PROPERTIES PREFIX "")
+    add_compile_definitions(BUILD_DLL) # source files should know whether they are built for dll or exe!
```

Comment:
another thing here:
there was a reason to add this definition to only one file using `set_source_files_properties` - that way you can have only one set of artifacts, and if you want to change build type (exe <-> dll) you can still use your build dir and will only need to recompile one .cpp (which actually needs this definition) and relink the final binary.  
meanwhile adding it for the whole target (or even globally, like here) will cause all the object files being recompiled for pretty much no reason. or one will have to keep two sets of artifacts, still for no good reason.  
as far as i can see you still need this definition in only two .cpp files, so i'd suggest to do it old way, just add a second file to the list